src/fan.ml

```diff
@@ -104,5 +106,8 @@ let import ~hash_size buf =
   in
   let size = int_of_float (log2 (float_of_int nb_fans)) in
   let shift = hash_size - size in
-  let mask = (nb_fans - 1) lsl shift in
+  let nb_fans = max 0 (nb_fans - 1) in
+  let mask = nb_fans lsl shift in
```

Comment:
I don't properly understand what's going on here, but I think `nb_fans` is a confusing name for `max 0 (nb_fans - 1)`.

Reply:
I wanted to avoid `nb_fans` and `mask` to become negative, but it seems that there are no repercussions on the tests either way so I removed it.